internal/ad/krb5.go

```diff
@@ -0,0 +1,66 @@
+package ad
+
+/*
+#include <errno.h>
+#include <string.h>
+
+#include <krb5.h>
+
+char *get_ticket_path() {
+  krb5_error_code ret;
+  krb5_context context;
+
+  ret = krb5_init_context(&context);
+  if (ret) {
+    errno = ret;
+    return NULL;
+  }
+
+  const char* cc_name = krb5_cc_default_name(context);
+  if (cc_name == NULL) {
+    return NULL;
+  }
+
+  return strdup(cc_name);
+}
+*/
+// #cgo pkg-config: krb5
+import "C"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"unsafe"
+
+	"github.com/leonelquinteros/gotext"
+	log "github.com/ubuntu/adsys/internal/grpc/logstreamer"
+	"golang.org/x/sys/unix"
+)
+
+// TicketPath returns the path of the default kerberos ticket cache for the
+// current user.
+// It returns an error if the path is empty or does not exist on the disk.
+func TicketPath(ctx context.Context) (string, error) {
+	cKrb5cc, err := C.get_ticket_path()
+	defer C.free(unsafe.Pointer(cKrb5cc))
+	if err != nil {
+		return "", err
+	}
+	krb5cc := C.GoString(cKrb5cc)
+	if krb5cc == "" {
+		return "", errors.New(gotext.Get("path is empty"))
+	}
+
+	krb5ccPath, found := strings.CutPrefix(krb5cc, "FILE:")
+	if !found {
+		log.Warningf(ctx, gotext.Get("Krb5cc path %q does not start with FILE:", krb5cc))
+	}
```

Comment:
Does this matter from our POV or does everything still work if the ticket is not a file? Sincere question.

Reply:
I don't think we should drop the check here, honestly. If the API returns something without the `FILE:` prefix, we should return an error instead of delegating the failure downstream, IMHO.

Looking at some docs (https://web.mit.edu/kerberos/krb5-1.12/doc/basic/ccache_def.html#ccache-types), there are quite some possible `ccache` types. Maybe we should cut the prefix and, if it's not `FILE` we return an error mentioning that prefix X is not supported by us?

Reply:
`access` would not fail for a dir, would it?